subprojects/core/src/main/java/org/gradle/initialization/ProjectPathRegistry.java

```diff
@@ -27,6 +27,18 @@
      */
     Set<Path> getAllProjectPaths();
 
+    /**
+     * Returns an path for every implicit project in a build, including projects from included builds.
+     * Always returns the same set for the lifetime of the build.
```

Comment:
I don't think this is true, and even if it is, it shouldn't be part of the contract. Implicit builds can be added at pretty much any time, and consumers of this interface should deal with this.